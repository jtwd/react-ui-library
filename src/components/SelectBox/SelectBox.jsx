import React from 'react'
import { string, oneOf, bool, func, node, arrayOf, object } from 'prop-types'

import Label from '../Label'
import { StyledSelectBox, Option } from './SelectBox.styles'
import { Field, ErrorMsg } from "../TextInput/TextInputStyles"

/** SelectBox field - Can be required, in error. Uses: Label, Icon. Options are passed in as an array of objects. Each option contains: id, label, value (optional). If value is not supplied, label is used as the value */
function SelectBox({ htmlId, size, name, label, required, defaultOptionLabel, onChange, options, defaultValue, error, children, ...props }) {
  if (options.length < 2) return null // there needs to be more than 1 option
  return (
    <Field size={size}>
      <Label htmlFor={htmlId} label={label} required={required} />
      <StyledSelectBox
        id={htmlId}
        name={name}
        onChange={onChange}
        error={error}
        value={defaultValue}
        {...props}
      >
        {defaultOptionLabel && <Option>{defaultOptionLabel}</Option>}
        {options && options.map((option) => {
          let optionVal = option.label
          if (option.value) optionVal = option.value
          return(
            <Option
              key={`${htmlId}-${option.id}`}
              value={optionVal}
            >{option.label}</Option>
          )
        })}
      </StyledSelectBox>
      {children}
      <ErrorMsg message={(error)} />
    </Field>
  )
}

SelectBox.propTypes = {
  /** ID for input to link with label */
  htmlId: string.isRequired,
  /** Name of input */
  name: string.isRequired,
  /** To go in label */
  label: string.isRequired,
  /** Size (default, xs, sm, md, lg) */
  size: oneOf(['default', 'xs', 'sm', 'md', 'lg', 'xl']),
  /** Required field */
  required: bool,
  /** Function to run when input is being typed in */
  onChange: func.isRequired,
  /** Value of select */
  // eslint-disable-next-line react/require-default-props
  defaultValue: string,
  /** Text to be shown in the first (default) option */
  defaultOptionLabel: string,
  /** Options for selectbox */
  options: arrayOf(object),
  /** Error message */
  error: string,
  /** TextInput contents */
  children: node
}

SelectBox.defaultProps = {
  size: 'default',
  required: false,
  defaultValue: '',
  defaultOptionLabel: 'Please select an option',
  error: null,
  options: [],
  children: null
}

export default SelectBox
