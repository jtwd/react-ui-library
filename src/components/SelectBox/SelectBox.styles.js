import styled, { css } from 'styled-components'

import { red, lightRed, darkRed, yellow, white, transBlackDark, transBlackLight, transWhite } from '../_theme/colors'
import { transitions, radiusRound } from '../_theme/units'
import { xs } from '../_theme/spacers'

export const StyledSelectBox = styled.select`
  border: 2px solid ${transWhite};
  background: ${transBlackLight};
  color: ${white};
  padding: ${xs};
  transition: ${transitions.hover};
  border-radius: ${radiusRound};
  display: block;
  width: 100%;
  height: 47px;

  ${props =>
    props.error &&
    css`
    border-color: ${lightRed};
  `} 
  
  &:hover {
    border-color: ${props => (props.error ? red : white)};
  }
  
  &:focus {
    border-color: ${props => (props.error ? darkRed : yellow)};
    background: ${transBlackDark};
    outline: 0;
  }
`

export const Option = styled.option`
  position: relative;
`
