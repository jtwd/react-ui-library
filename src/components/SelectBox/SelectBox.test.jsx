import React from 'react'
import renderer from 'react-test-renderer'

import SelectBox from './SelectBox'

test('SelectBox renders correctly', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox renders label as the option value when value is not supplied', () => {
  const options = [
    { id: 1, label: 'Option number 1' },
    { id: 2, label: 'Secondary option' },
    { id: 3, label: 'Option three' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox renders correctly when passed the required prop', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} required onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox renders correctly when passed the error prop', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} required error="This is required" onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox renders correctly when passed the defaultOptionLabel prop', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} defaultOptionLabel="Custom default option label" onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox does not render the default option when defaultOptionValue is null', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} defaultOptionLabel={null} onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox renders the correct option as selected when passed the value prop', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' },
    { id: 2, label: 'Secondary option', value: '4324344' },
    { id: 3, label: 'Option three', value: '4324366' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} value="4324344" onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox does not render if passed less than 2 options', () => {
  const options = [
    { id: 1, label: 'Option number 1', value: '4324324' }
  ]
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" options={options} onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SelectBox does not render if passed no options', () => {
  const component = renderer.create(<SelectBox htmlId="test" name="test" label="Test" onChange={() => {}} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
