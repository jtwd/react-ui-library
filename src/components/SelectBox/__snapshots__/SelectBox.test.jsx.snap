// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox does not render if passed less than 2 options 1`] = `null`;

exports[`SelectBox does not render if passed no options 1`] = `null`;

exports[`SelectBox does not render the default option when defaultOptionValue is null 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`SelectBox renders correctly 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Please select an option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`SelectBox renders correctly when passed the defaultOptionLabel prop 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Custom default option label
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`SelectBox renders correctly when passed the error prop 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
    <span
      className="sc-bwzfXH dEIIn"
      title="Required field"
    />
  </label>
  <select
    className="sc-htpNat chAfyq"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Please select an option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX iDFqMw sc-gqjmRU hEKEbc sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
      This is required
    </div>
  </div>
</div>
`;

exports[`SelectBox renders correctly when passed the required prop 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
    <span
      className="sc-bwzfXH dEIIn"
      title="Required field"
    />
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Please select an option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`SelectBox renders label as the option value when value is not supplied 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value={undefined}
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Please select an option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="Option number 1"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="Secondary option"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="Option three"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`SelectBox renders the correct option as selected when passed the value prop 1`] = `
<div
  className="sc-jzJRlG kZNvDb"
  size="default"
>
  <label
    className="sc-bdVaJa etThJd"
    htmlFor="test"
  >
    Test
     
  </label>
  <select
    className="sc-htpNat QxoBT"
    id="test"
    name="test"
    onChange={[Function]}
    value="4324344"
  >
    <option
      className="sc-bxivhb cdiCYE"
    >
      Please select an option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324324"
    >
      Option number 1
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324344"
    >
      Secondary option
    </option>
    <option
      className="sc-bxivhb cdiCYE"
      value="4324366"
    >
      Option three
    </option>
  </select>
  <div
    className="sc-fjdhpX dEJYHl sc-gqjmRU lhmBJm sc-EHOje jfQQfx sc-ifAKCX hShZxT"
  >
    <div
      className="Panel-body sc-bZQynM lfLAEX"
    >
      <i
        className="sc-dnqmqq huRoGJ"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m2.9 27.8v-4.2q0-0.4-0.2-0.6t-0.5-0.2h-4.3q-0.3 0-0.5 0.2t-0.2 0.5v4.3q0 0.3 0.2 0.5t0.5 0.2h4.3q0.3 0 0.5-0.2t0.2-0.5z m0-7.7l0.4-13.8q0-0.3-0.3-0.5-0.2-0.1-0.5-0.1h-4.9q-0.3 0-0.5 0.1-0.3 0.2-0.3 0.4l0.4 13.9q0 0.2 0.2 0.4t0.6 0.2h4.1q0.3 0 0.5-0.2t0.3-0.4z"
            />
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;
